client/src/app/shared/metadata-form/fields/ontology-input/ontology-input.component.ts

```diff
@@ -0,0 +1,122 @@
+import {Component, OnInit} from '@angular/core';
+import {AbstractControl, FormControl} from '@angular/forms';
+import {concatMap, startWith} from 'rxjs/operators';
+import {Observable} from 'rxjs';
+import {Metadata} from '../../metadata';
+import {OntologyService} from '../../../services/ontology.service';
+import {OlsDoc} from '../../../models/ols';
+import {JsonSchema} from '../../json-schema';
+
+const OLS_RELATION = {
+  'rdfs:subClassOf': 'allChildrenOf'
+};
+
+interface Ontology {
+  text: string;
+  ontology_label: string;
+  ontology: string;
+}
+
+
+@Component({
+  selector: 'app-ontology-input',
+  templateUrl: './ontology-input.component.html',
+  styleUrls: ['./ontology-input.component.css']
+})
+export class OntologyInputComponent implements OnInit {
+  metadata: Metadata;
+  control: AbstractControl;
+  id: string;
+
+  label: string;
+  helperText: string;
+  isRequired: boolean;
+  error: string;
+  example: string;
+  disabled: boolean;
+  value: Ontology;
+
+  searchControl: FormControl;
+  filteredOptions: Observable<Ontology[]>;
```

Comment:
Should this be `Observable<Ontology>`? Otherwise it looks like the we're passing `Ontology[]` to `displayOntology()` in the template, but it's type signature expects just an `Ontology`

Reply:
the filteredOptions is a list of Ontology objects (contains label and obo id), the displayOntology is formatting an Ontology object which is the value of each options.

Reply:
Just looked at *ngFor, seems like the "async" unrolls the `Observable<Ontology[]>` into just `Ontology[]`